Crash on 1.19.4

```

Reported exception thrown!
net.minecraft.class_148: Rendering screen
	at net.minecraft.class_757.method_3192(class_757.java:946)
	at net.minecraft.class_310.method_1523(class_310.java:1218)
	at net.minecraft.class_310.method_1514(class_310.java:801)
	at net.minecraft.client.main.Main.main(Main.java:237)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:462)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
Caused by: java.lang.NoSuchMethodError: 'void net.minecraft.class_2588.<init>(java.lang.String, java.lang.Object[])'
	at net.mision_thi.nbttooltips.tooltips.TooltipChanger.Main(TooltipChanger.java:25)
	at net.minecraft.class_1799.handler$coc000$nbttooltips$injectEditTooltipmethod(class_1799.java:4161)
	at net.minecraft.class_1799.method_7950(class_1799.java:834)
	at net.minecraft.class_481.method_25409(class_481.java:754)
	at net.minecraft.class_465.method_2380(class_465.java:179)
	at net.minecraft.class_481.method_25394(class_481.java:743)
	at net.minecraft.class_437.method_47413(class_437.java:121)
	at net.minecraft.class_757.method_3192(class_757.java:937)
	... 6 more
```

--- a/gradle.properties
+++ b/gradle.properties
@@ -3,14 +3,14 @@ org.gradle.jvmargs=-Xmx1G
 
 # Fabric Properties
 	# check these on https://fabricmc.net/develop
-	minecraft_version=1.19.4
-	yarn_mappings=1.19.4+build.2
-	loader_version=0.14.19
+	minecraft_version=1.20.1
+	yarn_mappings=1.20.1+build.2
+	loader_version=0.14.21
 
 # Mod Properties
-	mod_version = 0.1.4-1.19.x
+	mod_version = 0.1.4-1.20.x
 	maven_group = net.mision_thi
 	archives_base_name = nbttooltips
 
 # Dependencies
-	fabric_version=0.78.0+1.19.4
+	fabric_version=0.83.1+1.20.1

--- a/gradle.properties
+++ b/gradle.properties
@@ -3,9 +3,9 @@ org.gradle.jvmargs=-Xmx1G
 
 # Fabric Properties
 	# check these on https://fabricmc.net/develop
-	minecraft_version=1.19.3
-	yarn_mappings=1.19.3+build.5
-	loader_version=0.14.13
+	minecraft_version=1.19.4
+	yarn_mappings=1.19.4+build.2
+	loader_version=0.14.19
 
 # Mod Properties
 	mod_version = 0.1.4-1.19.x
@@ -13,4 +13,4 @@ org.gradle.jvmargs=-Xmx1G
 	archives_base_name = nbttooltips
 
 # Dependencies
-	fabric_version=0.73.2+1.19.3
+	fabric_version=0.78.0+1.19.4

--- a/src/main/java/net/mision_thi/nbttooltips/tooltips/TooltipChanger.java
+++ b/src/main/java/net/mision_thi/nbttooltips/tooltips/TooltipChanger.java
@@ -2,11 +2,10 @@ package net.mision_thi.nbttooltips.tooltips;
 
 import net.minecraft.client.MinecraftClient;
 import net.minecraft.item.ItemStack;
-import net.minecraft.text.*;
+import net.minecraft.text.MutableText;
+import net.minecraft.text.Text;
 import net.minecraft.util.Formatting;
 import net.mision_thi.nbttooltips.config.ModConfigs;
-import net.minecraft.text.Text;
-import net.minecraft.text.TranslatableTextContent;
 
 import java.util.ArrayList;
 import java.util.regex.Matcher;
@@ -22,7 +21,7 @@ public class TooltipChanger {
             With that NBT text recreated, we can find the index of the text.
             After we found it we remove it and keep the index stored in the `index` variable.
         */
-        temp.add(MutableText.of(new TranslatableTextContent("item.nbt_tags", itemStack.getNbt().getKeys().size())).formatted(Formatting.DARK_GRAY));
+        temp.add(Text.translatable("item.nbt_tags", itemStack.getNbt().getKeys().size()).formatted(Formatting.DARK_GRAY));
         int index = list.indexOf(temp.get(0));
         int indexInsertLocation = list.indexOf(temp.get(0));
         list.remove(index);
@@ -36,8 +35,8 @@ public class TooltipChanger {
         Matcher m = p.matcher(nbtList);
 
         // Create new literalText, which we will be adding to the list.
-        MutableText mutableText = MutableText.of(new LiteralTextContent(""));
-        mutableText.append(MutableText.of(new TranslatableTextContent("item.nbt_tags.nbttooltips")).formatted(Formatting.DARK_GRAY));
+        MutableText mutableText = Text.empty();
+        mutableText.append(Text.translatable("item.nbt_tags.nbttooltips").formatted(Formatting.DARK_GRAY));
 
 
         /*  **Loop through the NBT data**
@@ -70,12 +69,12 @@ public class TooltipChanger {
              */
             if (nbtList.charAt(m.start()) == '\'') {
                 if (singleQuotationMark.equals(Boolean.FALSE)) { // If false color only the quotation mark
-                    mutableText.append(MutableText.of(new LiteralTextContent(String.valueOf(nbtList.charAt(m.start())))).formatted(quotationColour));
+                    mutableText.append(Text.literal(String.valueOf(nbtList.charAt(m.start()))).formatted(quotationColour));
                     singleQuotationMark = Boolean.TRUE;
                 }
                 else { // Else color the quotation mark and make the rest green
-                    mutableText.append(MutableText.of(new LiteralTextContent(nbtList.substring(lastIndex+1,m.start()))).formatted(stringColour));
-                    mutableText.append(MutableText.of(new LiteralTextContent(String.valueOf(nbtList.charAt(m.start())))).formatted(quotationColour));
+                    mutableText.append(Text.literal(nbtList.substring(lastIndex+1,m.start())).formatted(stringColour));
+                    mutableText.append(Text.literal(String.valueOf(nbtList.charAt(m.start()))).formatted(quotationColour));
                     singleQuotationMark = Boolean.FALSE;
                 }
                 lastString = String.valueOf(nbtList.charAt(m.start()));
@@ -97,7 +96,7 @@ public class TooltipChanger {
                         Stores the lastString and lastIndex
                  */
                 if (nbtList.charAt(m.start()) == '{' || nbtList.charAt(m.start()) == '[' ) {
-                    mutableText.append(MutableText.of(new LiteralTextContent(String.valueOf(nbtList.charAt(m.start())))).formatted(separationColour));
+                    mutableText.append(Text.literal(String.valueOf(nbtList.charAt(m.start()))).formatted(separationColour));
                     lastString = String.valueOf(nbtList.charAt(m.start()));
                     lastIndex = m.start();
                 }
@@ -117,17 +116,17 @@ public class TooltipChanger {
                             nbtList.charAt(m.start()-1) == 'l' || nbtList.charAt(m.start()-1) == 'L' ||
                             nbtList.charAt(m.start()-1) == 'f' || nbtList.charAt(m.start()-1) == 'F'
                     ) {
-                        mutableText.append(MutableText.of(new LiteralTextContent(nbtList.substring(lastIndex+1,m.start()-1))).formatted(integerColour));
-                        mutableText.append(MutableText.of(new LiteralTextContent(nbtList.substring(m.start()-1,m.start()))).formatted(typeColour));
+                        mutableText.append(Text.literal(nbtList.substring(lastIndex+1,m.start()-1)).formatted(integerColour));
+                        mutableText.append(Text.literal(nbtList.substring(m.start()-1,m.start())).formatted(typeColour));
 
                     }
                     else {
-                        mutableText.append(MutableText.of(new LiteralTextContent(nbtList.substring(lastIndex+1,m.start()))).formatted(integerColour));
+                        mutableText.append(Text.literal(nbtList.substring(lastIndex+1,m.start())).formatted(integerColour));
                     }
 
-                    mutableText.append(MutableText.of(new LiteralTextContent(String.valueOf(nbtList.charAt(m.start()))))).formatted(separationColour);
+                    mutableText.append(Text.literal(String.valueOf(nbtList.charAt(m.start())))).formatted(separationColour);
 
-                    if (nbtList.charAt(m.start()) == ',') { mutableText.append(MutableText.of(new LiteralTextContent(" ")).formatted(separationColour)); }
+                    if (nbtList.charAt(m.start()) == ',') { mutableText.append(Text.literal(" ").formatted(separationColour)); }
                     lastString = String.valueOf(nbtList.charAt(m.start()));
                     lastIndex = m.start();
                 }
@@ -142,10 +141,10 @@ public class TooltipChanger {
                  */
                 if (nbtList.charAt(m.start()) == ':') { // 4).
                     if (!lastString.equals("\"")) {
-                        mutableText.append(MutableText.of(new LiteralTextContent(nbtList.substring(lastIndex+1,m.start()))).formatted(fieldColour));
+                        mutableText.append(Text.literal(nbtList.substring(lastIndex+1,m.start())).formatted(fieldColour));
 
-                        mutableText.append((MutableText.of(new LiteralTextContent(String.valueOf(nbtList.charAt(m.start()))))).formatted(separationColour));
-                        mutableText.append(MutableText.of(new LiteralTextContent(" ")).formatted(separationColour));
+                        mutableText.append((Text.literal(String.valueOf(nbtList.charAt(m.start())))).formatted(separationColour));
+                        mutableText.append(Text.literal(" ").formatted(separationColour));
                         lastString = String.valueOf(nbtList.charAt(m.start()));
                         lastIndex = m.start();
                     }
@@ -165,17 +164,17 @@ public class TooltipChanger {
 
                         // Check if the string is way too long
                         if (m.start() - lineLimit > lineStep ){
-                            mutableText.append(MutableText.of(new LiteralTextContent("....")).formatted(lstringColour));
+                            mutableText.append(Text.literal("....").formatted(lstringColour));
                             removedCharters += m.start() - lineLimit;
                         }
                         else {
-                            mutableText.append(MutableText.of(new LiteralTextContent(nbtList.substring(lastIndex+1,m.start()))).formatted(stringColour));
+                            mutableText.append(Text.literal(nbtList.substring(lastIndex+1,m.start())).formatted(stringColour));
                         }
 
-                        mutableText.append(MutableText.of(new LiteralTextContent(String.valueOf(nbtList.charAt(m.start())))).formatted(quotationColour));
+                        mutableText.append(Text.literal(String.valueOf(nbtList.charAt(m.start()))).formatted(quotationColour));
                     }
                     else {
-                        mutableText.append(MutableText.of(new LiteralTextContent(String.valueOf(nbtList.charAt(m.start())))).formatted(quotationColour));
+                        mutableText.append(Text.literal(String.valueOf(nbtList.charAt(m.start()))).formatted(quotationColour));
                     }
                     lastString = String.valueOf(nbtList.charAt(m.start()));
                     lastIndex = m.start();
@@ -194,14 +193,14 @@ public class TooltipChanger {
                 if (nbtList.charAt(m.start()) == '}' || nbtList.charAt(m.start()) == ']' || nbtList.charAt(m.start()) == ',') { // 2).
 
                     if (lastString.equals("'")) { // 3).
-                        mutableText.append(MutableText.of(new LiteralTextContent(nbtList.substring(lastIndex+1,m.start()))).formatted(stringColour));
+                        mutableText.append(Text.literal(nbtList.substring(lastIndex+1,m.start())).formatted(stringColour));
                         lastIndex = m.start();
                     }
 
                     // 4).
                     list.add(indexInsertLocation,mutableText);
                     indexInsertLocation += 1;
-                    mutableText = MutableText.of(new LiteralTextContent("     "));
+                    mutableText = Text.literal("     ");
                     lineAdded = Boolean.TRUE;
                     lineLimit = lineLimit + lineStep;
                 }

--- a/src/main/resources/assets/nbttooltips/lang/en_us.json
+++ b/src/main/resources/assets/nbttooltips/lang/en_us.json
@@ -1,4 +1,4 @@
 {
   "nbttooltips.keybind": "Show NBT",
-  "item.nbt_tags.nbttooltips": "NBT: %s"
+  "item.nbt_tags.nbttooltips": "NBT: "
 }

--- a/src/main/resources/fabric.mod.json
+++ b/src/main/resources/fabric.mod.json
@@ -1,7 +1,7 @@
 {
   "schemaVersion": 1,
   "id": "nbttooltips",
-  "version": "0.1.4-1.19.x",
+  "version": "0.1.4-1.20.x",
 
   "name": "NBT tooltips",
   "description": "A simple fabric Minecraft mod that adds a feature that allows you to see the custom NBT data when you hover over it and press your shift key.",
@@ -32,7 +32,7 @@
   "depends": {
     "fabricloader": ">=0.13.3",
     "fabric": "*",
-    "minecraft": "1.19.x",
+    "minecraft": "1.20.x",
     "java": ">=17",
     "fabric-key-binding-api-v1": "*"
   },

